packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -1341,3 +1305,30 @@ class AndroidDevicePortForwarder extends DevicePortForwarder {
     }
   }
 }
+
+// In platform tools 29.0.0 adb.exe seems to be ending with this heap
+// corruption error code on seemingly successful termination. Ignore
+// this error on windows.
+bool _allowHeapCorruptionOnWindows(int exitCode, Platform platform) {
+  return exitCode == -1073740940 && platform.isWindows;
+}
+
+/// Whether the [AndroidDevice] is believed to be a physical device or an emulator.
+enum HardwareType { emulator, physical }
```

Comment:
Made these public and documented on AndroidDevice. Just constants, should update the tests we have to use them instead of using specific strings